src/actions/tokens.js

```diff
@@ -0,0 +1,36 @@
+import { createActions } from 'redux-actions'
+import { getLikelyContracts, getMetadata, getBalanceOf } from '../utils/tokens'
+
+const WHITELISTED_CONTRACTS = (process.env.TOKEN_CONTRACTS || 'berryclub.ek.near,farm.berryclub.ek.near,wrap.near').split(',');
+
+export const handleGetTokens = () => async (dispatch, getState) => {
+    const { accountId } = getState().account
+
+    await dispatch(tokens.likelyContracts.get(accountId))
+
+    const { tokens: contracts } = getState().tokens
+
+    await Promise.all(Object.keys(contracts).map(async contract => {
+        await dispatch(tokens.tokensDetails.getMetadata(contract, accountId))
+    }))
+
+    Object.keys(contracts).map(async contract => {
+        await dispatch(tokens.tokensDetails.getBalanceOf(contract, accountId))
+    })
+}
+
+export const { tokens } = createActions({
+    TOKENS: {
+        LIKELY_CONTRACTS: {
+            GET: [
+                getLikelyContracts,
+                () => WHITELISTED_CONTRACTS
+            ],
+        },
+        TOKENS_DETAILS: {
+            GET_METADATA: getMetadata,
+            GET_BALANCE_OF: getBalanceOf
+        },
+        CLEAR_STATE: null
```

Comment:
Given there are so few keys and the intention of the actions can easily be described without nesting -- why not define these in a flat fashion like e.g. `getLikelyTokenNames`, `getTokenMetadata` and `getTokenBalances`?